src/arcsf/data/data_utils.py

```diff
@@ -0,0 +1,244 @@
+import itertools
+
+import numpy as np
+from datasets import Dataset, load_dataset
+
+
+def get_indices_structured(
+    forgotten_author_numbers: list,
+    q_remove: int = 4,
+    n_authors: int = 200,
+    q_per_author: int = 20,
+):
+    """
+    Returns two flattened lists of Question--Answer indices given authors numbers to
+    remove/retain, in the case where the forget target can be easily structured.
+    Assumes equal questions per author, and sorted authors.
+
+        Parameters:
+                forgotten_author_numbers (list):    List of integers containing author
+                                                    indexes about which facts should
+                                                    be removed
+                n_authors (int):                    Total number of authors
+                q_removal (int):                    Number of questions/facts that
+                                                    should be removed
+                q_per_author (int):                 Integer defining the number of
+                                                    questions to remove per author
+
+        Returns:
+                forget_indices (list):              List of integers corresponding to
+                                                    the question indices pertaining the
+                                                    authors in author_numbers
+                retain_indices (list):              List of integers corresponding to
+                                                    the remaining question indices
+    """
+    assert (
+        q_remove <= q_per_author
+    ), "Cannot remove more questions for an author than there are questions per author."
```

Comment:
Checks like this generally should be replaced with raising exceptions rather than using `assert` (as `assert` can be ignored at runtime)